src/modules/rover_pos_control/RoverPositionControl.cpp

```diff
@@ -419,15 +428,25 @@ RoverPositionControl::control_attitude(const vehicle_attitude_s &att, const vehi
 {
 	// quaternion attitude control law, qe is rotation from q to qd
 	const Quatf qe = Quatf(att.q).inversed() * Quatf(att_sp.q_d);
+
+	// Attitude error in Euler angles [rad]
 	const Eulerf euler_sp = qe;
-	// PX4_INFO("Yaw error: %f", double(euler_sp(2)));
+	// PX4_INFO("Yaw error: %f [rad]", double(euler_sp(2)));
 
 	_rates_sp.roll = 0.0;
 	_rates_sp.pitch = 0.0;
-	_rates_sp.yaw = euler_sp(2) / _param_max_turn_angle.get();
+
+	// Apply Attitude P-gain for Yaw control. Euler angle in [rad], rate setpoint in [rad/s]
+	_rates_sp.yaw = euler_sp(2) * _param_att_p.get();
+
+	// PX4_INFO("Yaw rate setpoint in attitude control: %f [rad/s]", (double)_rates_sp.yaw);
+
 	_rates_sp.thrust_body[0] = math::constrain(att_sp.thrust_body[0], 0.0f, 1.0f);
 	_rates_sp.timestamp = hrt_absolute_time();
 	_rates_sp_pub.publish(_rates_sp);
+
+	// Log yaw setpoint for PID tuning debug
+	_local_pos_setpoint.yaw = att_sp.yaw_body;
```

Comment:
@Jaeyoung-Lim I actually find it quite confusing / weird that we keep both euler angle & quaternion info inside the attitude setpoint data structure :thinking:. But that's the way it is defined for now.